apps/ledger-live-mobile/src/components/Nft/NftList.hook.ts

```diff
@@ -37,9 +38,26 @@ export function useNftList() {
   // Detect change of screen
   useEffect(() => {
     if (!isFocused) {
-      readOnlyModeAction();
+      exitMultiSelectMode();
     }
-  }, [readOnlyModeAction, isFocused]);
+
+    return () => {
+      exitMultiSelectMode();
+    };
+  }, [exitMultiSelectMode, isFocused, dispatch]);
+
+  const handleBackPress = useCallback(() => {
+    exitMultiSelectMode();
```

Comment:
Here you have to handle the case where you're not in multiSelectMode and then return false (so that other eventual other listeners (somewhere else in the app) can do their job, for instance the default native one which will trigger an app close in that case)